pyproject.toml

```
@@ -0,0 +1,80 @@
+[build-system]
+requires = ["flit_core >=3.2,<4"]
+build-backend = "flit_core.buildapi"
+
+[project]
+name = "py2spack"
+authors = [{name = "davhofer", email = "davhofer@ethz.ch"}]
+readme = "README.md"
+license = {text = "MIT"}
+classifiers = ["License :: OSI Approved :: MIT License"]
+dynamic = ["version", "description"]
+dependencies = [
+    "packaging>=20", 
+    "requests>=2.30", 
+    "tomli>=1.2",
+]
+
+[project.urls]
+Home = "https://github.com/davhofer/py2spack/"
+
+
+[tool.ruff]
+    exclude = [
+    ".bzr",
+    ".direnv",
+    ".eggs",
+    ".git",
+    ".git-rewrite",
+    ".hg",
+    ".ipynb_checkpoints",
+    ".mypy_cache",
+    ".nox",
+    ".pants.d",
+    ".pyenv",
+    ".pytest_cache",
+    ".pytype",
+    ".ruff_cache",
+    ".svn",
+    ".tox",
+    ".venv",
+    ".vscode",
+    "__pypackages__",
+    "_build",
+    "buck-out",
+    "build",
+    "dist",
+    "node_modules",
+    "site-packages",
+    "venv",
+]
+
+line-length = 80
+indent-width = 4
+
+[tool.ruff.lint]
+select = ["E", "F", "W", "B", "N", "D"]
```

I prefer to leave the default selected rules and add many other useful ones. If this configuration is too pedantic, some sets could be drop and some specific rules ignored
```suggestion
extend-safe-fixes = ["D", "TCH"]
extend-select = [
  "A",  # flake8-builtins
  "B",  # flake8-bugbear
  "I",  # isort
  "G",  # flake8-logging-format
  "N",  # pep8-naming
  "W",  # pycodestyle-warning
  "C4",  # flake8-comprehensions
  "C90",  # mccabe
  "D",  # pydocstyle
  "D213",  # multi-line-summary-second-line (off by default in pydocstyle "google' convention)
  "PT",  # flake8-pytest-style
  "TD",  # flake8-todo
  "UP",  # pyupgrade
  "ARG",  # flake8-unused-arguments
  "ERA",  # eradicate
  "FLY",  # flynt
  "ICN",  # flake8-import-conventions
  "PERF",  # Perflint
  "PGH",  # pygrep-hooks
  "PIE",  # flake8-pie
  "PL",  # pylint
  "PTH",  # flake8-use-pathlib
  "RET",  # flake8-return
  "RUF",  # Ruff-specific
  "SIM",  # flake8-simplify
  "T10",  # flake8-debugger
  "T20",  # flake8-print
  "TCH",  # flake8-type-checking
]
```